eslint.config.mjs

```diff
@@ -0,0 +1,131 @@
+/**
+ * SPDX-FileCopyrightText: 2022 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+// eslint.config.js
+
+import { recommendedVue2 } from '@nextcloud/eslint-config'
+import globals from 'globals'
+
+export default [
+	// Nextcloud ESLint config
+	...recommendedVue2,
+	// Ignore Nextcloud Server styles - requested from server
+	{
+		ignores: ['src/shared/renderer/assets/**/*'],
+	},
+	// Configuration
+	{
+		languageOptions: {
+			globals: {
+				// Electron Forge build vars
+				AUTHENTICATION_WINDOW_WEBPACK_ENTRY: 'readonly',
+				AUTHENTICATION_WINDOW_PRELOAD_WEBPACK_ENTRY: 'readonly',
+				CALLBOX_WINDOW_PRELOAD_WEBPACK_ENTRY: 'readonly',
+				CALLBOX_WINDOW_WEBPACK_ENTRY: 'readonly',
+				TALK_WINDOW_WEBPACK_ENTRY: 'readonly',
+				TALK_WINDOW_PRELOAD_WEBPACK_ENTRY: 'readonly',
+				HELP_WINDOW_WEBPACK_ENTRY: 'readonly',
+				HELP_WINDOW_PRELOAD_WEBPACK_ENTRY: 'readonly',
+				UPGRADE_WINDOW_WEBPACK_ENTRY: 'readonly',
+				UPGRADE_WINDOW_PRELOAD_WEBPACK_ENTRY: 'readonly',
+				WELCOME_WINDOW_WEBPACK_ENTRY: 'readonly',
+				WELCOME_WINDOW_PRELOAD_WEBPACK_ENTRY: 'readonly',
+				// Build constants
+				IS_DESKTOP: 'readonly',
+				__IS_DESKTOP__: 'readonly',
+				__CHANNEL__: 'readonly',
+				__VERSION_TAG__: 'readonly',
+				__TALK_VERSION_TAG__: 'readonly',
+				// Electron
+				...globals.node,
+			},
+		},
+		settings: {
+			jsdoc: {
+				tagNamePreference: {
+					return: 'return', // TODO: upstream
+				},
+			},
+		},
+	},
+	// Rules and overrides
+	{
+		rules: {
+			/**
+			 * ESLint
+			 */
+			'no-console': 'off', // TODO: remove after preview
+			/**
+			 * Nextcloud
+			 */
+			// Talk Desktop doesn't use real Nextcloud server globals
+			'@nextcloud/no-deprecations': 'off',
+			'@nextcloud/no-removed-apis': 'off',
+		},
+	},
+	// TODO: upstream
+	{
+		rules: {
+			'jsdoc/check-tag-names': ['error', { typed: false }], // TODO: upstream
+			'jsdoc/no-types': 'off', // TODO: upstream
+			'@stylistic/newline-per-chained-call': 'off', // TODO: upstream
+			'@stylistic/array-bracket-newline': ['error', { multiline: true }], // TODO: upstream
+			'@stylistic/array-element-newline': ['error', 'consistent'], // TODO: upstream
+			'@stylistic/object-property-newline': 'off', // TODO: upstream
+			'prefer-template': 'off', // TODO: upstream
+			'no-use-before-define': ['error', { functions: false }], // TODO: upstream
+		},
+	},
+	{
+		files: ['**/*.ts', '**/*.vue'],
+		rules: {
+			'@typescript-eslint/no-shadow': 'off', // TODO: upstream
+			'@typescript-eslint/no-use-before-define': ['error', { functions: false }], // TODO: upstream
+		},
+	},
+	{
+		files: ['**/*.vue'],
+		rules: {
+			'vue/require-prop-comment': 'off', // TODO: upstream
+		},
+	},
```

Comment:
Only these rules are "new".

All the other parts in the file are the old config migrated to a new Flat config format